### ISSUE TITLE
add dry run option to upload/download

### ISSUE THREAD

**Issue by satra** [dandi/dandi-cli#593]
it would be nice to be able to check before uploading/downloading what files would be updated/changed.

**Comment by yarikoptic** [comment]
yeah.  If too intrusive to implement ATM, may be should be done after new Python API RFing is settled (#249) and implemented, WDYT @jwodder ?

Note: the only "dry" mode we have ATM is for organize and it is invoked as `dandi organize --files-mode dry`. But I guess for consistency it could become a dedicated flag (`-n|--dry-mode` or just `--dry`?) not a `--files-mode`, and announce on actions to be done for specific mode.

**Comment by jwodder** [comment]
I'd prefer to do this after the new Python API gets at least somewhat implemented.

**Comment by yarikoptic** [comment]
@jwodder please have a fresh look since new Python API is already in place.
